portal_backend/views/organization.py

```diff
@@ -194,24 +194,25 @@ def post(self, request: Request, org_slug: str) -> Response:
 
         except User.DoesNotExist:
             user = User(email=email, role="maintainer")
-            user.full_clean()
-            user.save()
-
-        except ValidationError as e:
-            transaction.set_rollback(True)
-            logger.error(f"Validation error creating user: {e}")
-            return Response(
-                {"error": e.message_dict},
-                status=status.HTTP_400_BAD_REQUEST,
-            )
-        except Exception as e:
-            transaction.set_rollback(True)
-            logger.error(f"Unexpected error creating user: {e}")
-            return Response(
-                {"error": "Failed to create user"},
-                status=status.HTTP_500_INTERNAL_SERVER_ERROR,
-            )
-        user.organizations.add(organization)
+            try:
+                user.full_clean()
+                user.save()
+                user.organizations.add(organization)
```

Comment:
A user that isn't associated with this organisation can never be added to the organisation. That's why the add organisation part was outside of the try-catch :)